docs/source/common/hyperparameters.rst

```diff
@@ -152,22 +152,25 @@ improve readability and reproducibility.
         model = LitMNIST.load_from_checkpoint(PATH, loss_fx=torch.nn.SomeOtherLoss, generator_network=MyGenerator())
 
 
-3.  You can also save full objects such as `dict` or `Namespace` to the checkpoint.
+3.  You can also convert full objects such as ``dict`` or ``Namespace`` to ``hparams`` so they get saved to the checkpoint.
 
     .. code-block:: python
 
-        # using a argparse.Namespace
         class LitMNIST(LightningModule):
 
-            def __init__(self, conf, *args, **kwargs):
+            def __init__(self, conf: Union[Dict, Namespace, DictConfig], *args, **kwargs):
                 super().__init__()
                 self.save_hyperparameters(conf)
```

Comment:
I just noticed, I think these docs are incomplete. While this works for purely saving the parameters, it will not be possible to restore the model because it's ambiguous how to pass arguments to the constructor. 

One should either unroll the config in the signature, i.e., `def __init__(**conf)` or save it like so: `self.save_hyperparameters("conf")`